pkg/domain/infra/abi/play.go

```diff
@@ -1001,6 +1009,24 @@ func (ic *ContainerEngine) playKubePVC(ctx context.Context, pvcYAML *v1.Persiste
 	return &report, nil
 }
 
+func mergePublishPorts(p *entities.PodCreateOptions, publishPortsOption []nettypes.PortMapping) {
+	for _, publishPortSpec := range p.Net.PublishPorts {
+		if !portAlreadyPublished(publishPortSpec.ContainerPort, publishPortsOption) {
+			publishPortsOption = append(publishPortsOption, publishPortSpec)
+		}
+	}
+	p.Net.PublishPorts = publishPortsOption
+}
+
+func portAlreadyPublished(port uint16, publishedPorts []nettypes.PortMapping) bool {
+	for _, publishedPort := range publishedPorts {
+		if port >= publishedPort.ContainerPort && port < publishedPort.ContainerPort+publishedPort.Range {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
this works better but ignores the protocol, if you have a udp port in the yaml and also forward tcp via cli it must use both ports
if you fix this please also add a test for this case

Reply:
Fixed and added two tests at the bottom